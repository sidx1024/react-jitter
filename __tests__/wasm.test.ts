import { expect, test } from "vitest";
import { transform } from "@swc/core";
import type { ParserConfig } from "@swc/core";
import path from "node:path";
import url from "node:url";
import fs from "node:fs";

const pluginName =
  process.env.TEST_DEBUG === "true"
    ? "swc_plugin_react_jitter_debug.wasm"
    : "swc_plugin_react_jitter.wasm";
const pluginPath = path.join(
  path.dirname(url.fileURLToPath(import.meta.url)),
  "..",
  "plugin-swc",
  pluginName
);

const transformCode = async (
  code: string,
  options = {},
  filename = "test.jsx"
) => {
  const ext = path.extname(filename);
  const isTypescript = ext === ".ts" || ext === ".tsx";
  const parser: ParserConfig = isTypescript
    ? { syntax: "typescript", tsx: ext === ".tsx" }
    : { syntax: "ecmascript", jsx: true };
  return transform(code, {
    jsc: {
      parser,
      target: "es2018",
      experimental: { plugins: [[pluginPath, options]] },
    },
    filename,
  });
};

const fixturesDir = path.join(
  path.dirname(url.fileURLToPath(import.meta.url)),
  "fixtures"
);
const fixtureFiles = fs
  .readdirSync(fixturesDir)
  .filter((f) => f.endsWith(".jsx") || f.endsWith(".tsx"));

for (const file of fixtureFiles) {
  test(`fixture: ${file}`, async () => {
    const input = fs.readFileSync(path.join(fixturesDir, file), "utf-8");
    const options = file === "1_default.tsx" ? { includeArguments: true } : {};
    const { code } = await transformCode(input, options, file);
    expect(code.trim()).toMatchSnapshot();
  });
}
