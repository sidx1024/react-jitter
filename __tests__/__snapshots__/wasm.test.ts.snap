// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixture: 1_default.tsx 1`] = `
"import { useState } from "react";
import { useJitterScope } from "react-jitter";
export default function UserForm() {
    const h = useJitterScope({
        name: "UserForm",
        id: "b492276b",
        file: "$DIR/1_default.tsx",
        line: 3,
        offset: 0
    });
    const [count, setCount] = (h.s("2aeb03d6"), h.e(useState(0), {
        id: "2aeb03d6",
        hook: "useState",
        line: 4,
        offset: 28
    }));
    const nameFieldValues = (h.s("0eef1433"), h.e(useFieldValues("name"), {
        id: "0eef1433",
        hook: "useFieldValues",
        line: 5,
        offset: 26
    }));
    const addressFieldValues = (h.s("9a686f37"), h.e(useFieldValues("address"), {
        id: "9a686f37",
        hook: "useFieldValues",
        line: 6,
        offset: 29
    }));
    return /*#__PURE__*/ React.createElement("div", null);
}"
`;

exports[`fixture: 2_arrow_function.tsx 1`] = `
"import { useJitterScope } from "react-jitter";
export const useAddressField = ()=>{
    const h = useJitterScope({
        name: "useAddressField",
        id: "27dbacf3",
        file: "useAddressField.tsx",
        line: 1,
        offset: 31
    });
    return h.s("41f1f9ce"), h.e(useFieldValues("address"), {
        id: "41f1f9ce",
        hook: "useFieldValues",
        line: 1,
        offset: 37
    });
};"
`;

exports[`fixture: 3_anonymous_function.tsx 1`] = `
"import { useJitterScope } from "react-jitter";
export default function() {
    const h = useJitterScope({
        name: "(anonymous)",
        id: "95297035",
        file: "$DIR/3_anonymous_function.tsx",
        line: 1,
        offset: 0
    });
    const fieldValues = (h.s("9f49aae6"), h.e(useFieldValues("name"), {
        id: "9f49aae6",
        hook: "useFieldValues",
        line: 2,
        offset: 22
    }));
    return /*#__PURE__*/ React.createElement("div", null);
}"
`;

exports[`fixture: 4_non_react_function.tsx 1`] = `
"import { useJitterScope } from "react-jitter";
export function notAReactComponent() {
    return 5;
}"
`;
